### PR TITLE
Topic/coverity fixes

### LINKED ISSUE #345: clean coverity defects
see below https://scan8.coverity.com/reports.htm#v27370/p11529

### PR DESCRIPTION
**Pull request by yosefe**
replaced #1116 
fixes #345

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/973/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2934/ for details (Mellanox internal link).

File: contrib/test_jenkins.sh [hunk 1]
```diff
@@ -379,7 +379,7 @@ run_coverity() {
 		cov_build="$WORKSPACE/$cov_build_id"
 		rm -rf $cov_build
 		cov-build   --dir $cov_build $MAKE all
-		cov-analyze --dir $cov_build
+		cov-analyze $COV_OPT --dir $cov_build
```
Comment (yosefe) [inline contrib/test_jenkins.sh]: it's defined by Mellanox jenkins system. It's value is:
"--enable-fnptr --fnptr-models --all --paths 20000"

Comment (shamisp) [inline contrib/test_jenkins.sh]: Shell we define an internal variable that is overwritten by external + documentation.

Comment (yosefe) [inline contrib/test_jenkins.sh]: currently we don't have any options so de facto we have empty var..
however i can add doc to test_jenkins.sh on which env vars affect the script

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/981/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2945/ for details (Mellanox internal link).

**Pull request merged**